recipes/fmt/all/test_cmakedeps/conanfile.py

```diff
@@ -0,0 +1,25 @@
+import os
+from conans import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps
+from conan import tools
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "CMakeDeps"
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["FMT_HEADER_ONLY"] = self.options["fmt"].header_only
```

Comment:
self.options["fmt"] shouldn't be used to access dependencies information, as it might not be defined.
Need to use ``self.dependencies["fmt"].options.header_only``